targets/TARGET_Analog_Devices/TARGET_ADUCM4X50/TARGET_ADUCM4050/api/analogin_api.c

```diff
@@ -0,0 +1,212 @@
+extern "C" {
+#endif
+
+/* ADC Device number */
+#define ADC_DEV_NUM                (0u)
+
+/* Memory Required for adc driver */
+static uint32_t DeviceMemory[(ADI_ADC_MEMORY_SIZE+3)/4];
+/* Active channel */
+static uint32_t adi_pin2channel(PinName pin);
+
+/**
+ * \defgroup hal_analogin Analogin hal functions
+ * @{
+ */
+
+/** Initialize the analogin peripheral
+ *
+ * Configures the pin used by analogin.
+ * @param obj The analogin object to initialize
+ * @param pin The analogin pin name
+ */
+void analogin_init(analogin_t *obj, PinName pin)
+{
+    ADI_ADC_HANDLE hDevice;
+    bool bCalibrationDone = false;
+    bool bReady = false;
+
+    ADCName peripheral;
+    uint32_t function, channel;
+
+    peripheral = (ADCName)pinmap_peripheral(pin, &PinMap_ADC[0]);	// gives peripheral
+    MBED_ASSERT(peripheral != (ADCName)NC);
+
+    /* verify read function */
+    function = pinmap_function(pin, &PinMap_ADC[0]);
+    MBED_ASSERT(function == 1);
+
+    /* Configure PORT2_MUX registers */
+    pin_function(pin, function);
+
+    /* Configure active channel */
+    channel = adi_pin2channel(pin);
+    MBED_ASSERT(channel != 0xFFFFFFFF);
+    obj->UserBuffer.nChannels = channel;
+
+    /* Open the ADC device */
+    adi_adc_Open(ADC_DEV_NUM, DeviceMemory, sizeof(DeviceMemory), &hDevice);
+    obj->hDevice = hDevice;
+
+    /* Power up ADC */
+    adi_adc_PowerUp(hDevice, true);
+
+    /* Set ADC reference */
+    adi_adc_SetVrefSource(hDevice, ADI_ADC_VREF_SRC_INT_2_50_V);
+
+    /* Enable ADC sub system */
+    adi_adc_EnableADCSubSystem(hDevice, true);
+
+    /* Wait untilthe ADC is ready for sampling */
+    while(bReady == false) {
+        adi_adc_IsReady(hDevice, &bReady);
+    }
+
+    /* Start calibration */
+    adi_adc_StartCalibration(hDevice);
+
+    /* Wait until calibration is done */
+    while (!bCalibrationDone) {
+        adi_adc_IsCalibrationDone(hDevice, &bCalibrationDone);
+    }
+
+    /* Set the delay time */
+    adi_adc_SetDelayTime(hDevice, obj->DelayCycles);
+
+    /* Set the acquisition time. (Application need to change it based on the impedence) */
+    adi_adc_SetAcquisitionTime(hDevice, obj->SampleCycles);
+
+}
+
+/** Read the input voltage, represented as a float in the range [0.0, 1.0]
+ *
+ * @param obj The analogin object
+ * @return A floating value representing the current input voltage
+ */
+float analogin_read(analogin_t *obj)
+{
+    float fl32 = (float)analogin_read_u16(obj)/(float)4095.0;
+
+    return(fl32);
+}
+
+/** Read the value from analogin pin, represented as an unsigned 16bit value
+ *
+ * @param obj The analogin object
+ * @return An unsigned 16bit value representing the current input voltage
+ */
+uint16_t analogin_read_u16(analogin_t *obj)
+{
+    ADI_ADC_HANDLE  hDevice = obj->hDevice;
+    ADI_ADC_BUFFER  *pAdcBuffer;
+
+    /* Submit the buffer to the driver */
+    adi_adc_SubmitBuffer(hDevice, &obj->UserBuffer);
+
+    /* Enable the ADC */
+    adi_adc_Enable(hDevice, true);
```

Comment:
If this enables ADC everytime it is read, is there any disable anywhere, or is this enable required after submiting buffer?

Reply:
The enable call is required after the submitbuffer call.